simplify stub gemspec and remove invalid dev dependencies ... allow r…

grosser opened the pull request:
…unning bundle exec puma locally

@nateberkopec 
/cc @schneems

grosser commented on Rakefile:
```diff
@@ -17,13 +18,9 @@ HOE = Hoe.spec "puma" do
-
-  dependency "rack", [">= 1.1", "< 3.0"], :development
-
-  extra_dev_deps << ["rake-compiler", "~> 0.8"]
```
these do not match what is in the Gemfile and used locally or travis ... useless anyway so just removing them ...

grosser commented on puma.gemspec:
```diff
@@ -1,52 +1,19 @@
-else
-  version = "0.0.1"
-end
+version = File.read(File.expand_path("../lib/puma/const.rb", __FILE__))[/VERSION = "(\d+\.\d+\.\d+)"/, 1] || raise
```
if this fails then something is wrong ... better fail and fix instead of having weird versions

grosser commented on puma.gemspec:
```diff
@@ -1,52 +1,19 @@
-    s.add_dependency(%q<rdoc>, ["~> 4.0"])
-    s.add_dependency(%q<rake-compiler>, ["~> 0.8.0"])
-    s.add_dependency(%q<hoe>, ["~> 3.6"])
-  end
```
don't need any of that to work locally or via git dependency in bundler

nateberkopec commented on puma.gemspec:
```diff
@@ -1,52 +1,19 @@
   s.homepage = "http://puma.io"
   s.license = "BSD-3-Clause"
-  s.rdoc_options = ["--main", "README.md"]
-  s.require_paths = ["lib"]
```
isn't this a required attribute?

nateberkopec replied:
looked it up, this is the default value, so 👍

nateberkopec commented on puma.gemspec:
```diff
@@ -1,52 +1,19 @@
   s.files = `git ls-files`.split($/)
   s.homepage = "http://puma.io"
   s.license = "BSD-3-Clause"
-  s.rdoc_options = ["--main", "README.md"]
```
why remove?

nateberkopec replied:
from what I can tell removing this would change the default output for `gem install --document puma`.

nateberkopec commented on puma.gemspec:
```diff
@@ -1,52 +1,19 @@
   s.license = "BSD-3-Clause"
-  s.rdoc_options = ["--main", "README.md"]
-  s.require_paths = ["lib"]
-  s.required_ruby_version = Gem::Requirement.new(">= 1.9.3")
```
why remove?

nateberkopec replied:
removing this would remove the require ruby version check from bundler AFAICT, I don't want to do that.

grosser commented:
not required.
removing all the things that are unnecessary

On Thu, Jun 1, 2017 at 5:09 PM, Nate Berkopec <notifications@github.com>
wrote:

> *@nateberkopec* requested changes on this pull request.
> ------------------------------
>
> In puma.gemspec
> <https://github.com/puma/puma/pull/1313#discussion_r119757583>:
>
> >    s.email = ["evan@phx.io"]
>    s.executables = ["puma", "pumactl"]
>    s.extensions = ["ext/puma_http11/extconf.rb"]
>    s.files = `git ls-files`.split($/)
>    s.homepage = "http://puma.io"
>    s.license = "BSD-3-Clause"
> -  s.rdoc_options = ["--main", "README.md"]
> -  s.require_paths = ["lib"]
>
> isn't this a required attribute?
> ------------------------------
>
> In puma.gemspec
> <https://github.com/puma/puma/pull/1313#discussion_r119757593>:
>
> >    s.email = ["evan@phx.io"]
>    s.executables = ["puma", "pumactl"]
>    s.extensions = ["ext/puma_http11/extconf.rb"]
>    s.files = `git ls-files`.split($/)
>    s.homepage = "http://puma.io"
>    s.license = "BSD-3-Clause"
> -  s.rdoc_options = ["--main", "README.md"]
>
> why remove?
> ------------------------------
>
> In puma.gemspec
> <https://github.com/puma/puma/pull/1313#discussion_r119757604>:
>
> >    s.email = ["evan@phx.io"]
>    s.executables = ["puma", "pumactl"]
>    s.extensions = ["ext/puma_http11/extconf.rb"]
>    s.files = `git ls-files`.split($/)
>    s.homepage = "http://puma.io"
>    s.license = "BSD-3-Clause"
> -  s.rdoc_options = ["--main", "README.md"]
> -  s.require_paths = ["lib"]
> -  s.required_ruby_version = Gem::Requirement.new(">= 1.9.3")
>
> why remove?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/puma/puma/pull/1313#pullrequestreview-41667693>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ0DNseI0l9YQ-2bH4z87VzkARCAiks5r_1K0gaJpZM4Ntqrs>
> .
>

grosser commented:
please let me know which of the removed options is needed to run tests/use as git dependency ... or why else we'd need them would like to make this hacky gemspec as simple as possible

grosser commented:
this is not the real gemspec, just a stub ... so nobody will use it for gem
install

On Fri, Jun 2, 2017 at 5:40 AM, Nate Berkopec <notifications@github.com>
wrote:

> *@nateberkopec* commented on this pull request.
> ------------------------------
>
> In puma.gemspec
> <https://github.com/puma/puma/pull/1313#discussion_r119845984>:
>
> >    s.email = ["evan@phx.io"]
>    s.executables = ["puma", "pumactl"]
>    s.extensions = ["ext/puma_http11/extconf.rb"]
>    s.files = `git ls-files`.split($/)
>    s.homepage = "http://puma.io"
>    s.license = "BSD-3-Clause"
> -  s.rdoc_options = ["--main", "README.md"]
>
> from what I can tell removing this would change the default output for gem
> install --document puma.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/puma/puma/pull/1313#discussion_r119845984>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZy1IcueHmY_siDQmROM9moS2d7Asks5sAALJgaJpZM4Ntqrs>
> .
>

grosser commented:
restored ruby requirement

On Fri, Jun 2, 2017 at 5:41 AM, Nate Berkopec <notifications@github.com>
wrote:

> *@nateberkopec* commented on this pull request.
> ------------------------------
>
> In puma.gemspec
> <https://github.com/puma/puma/pull/1313#discussion_r119846108>:
>
> >    s.email = ["evan@phx.io"]
>    s.executables = ["puma", "pumactl"]
>    s.extensions = ["ext/puma_http11/extconf.rb"]
>    s.files = `git ls-files`.split($/)
>    s.homepage = "http://puma.io"
>    s.license = "BSD-3-Clause"
> -  s.rdoc_options = ["--main", "README.md"]
> -  s.require_paths = ["lib"]
> -  s.required_ruby_version = Gem::Requirement.new(">= 1.9.3")
>
> removing this would remove the require ruby version check from bundler
> AFAICT, I don't want to do that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/puma/puma/pull/1313#discussion_r119846108>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ46LnuHnfgzZCMwFVjywHxXQK9S6ks5sAALxgaJpZM4Ntqrs>
> .
>

grosser commented:
@nateberkopec good to go ?
